custom_components/versatile_thermostat/underlyings.py

```diff
@@ -1250,11 +1251,20 @@ async def start_cycle(
         # if force:
         await self.set_valve_open_percent()
 
+    @overrides
+    async def turn_off(self):
+        _LOGGER.debug("%s - ValveRegulation turn_off -> enforce min_opening_degree", self)
+        self._percent_open = 0
+        await self.send_percent_open()
+
     @property
     def is_device_active(self):
         """If the opening valve is open."""
         try:
-            return get_safe_float(self._hass, self._opening_degree_entity_id) > 0
+            #return get_safe_float(self._hass, self._opening_degree_entity_id) > 0
+            if self.hvac_mode == HVACMode.OFF:
+                return False
+            return self._thermostat.valve_open_percent > 0
```

Comment:
When doing that you check if the valve should be open and not if it is really open. I'm not fan of this change.